src/main/java/javaslang/control/Option.java

```diff
@@ -53,6 +53,19 @@
     }
 
     /**
+     *
+     * Creates {@code Some} of suppliers value if condition is true, or {@code None} in other case
+     *
+     * @return return {@code Some} of supplier's value if condition is true, or {@code None} in other case
+     */
+    static <T> Option<T> when(boolean condition, Supplier<T> supplier) {
```

Comment:
Let's use `Supplier<? extends T>`. There are situations where it is necessary.

Also we do null checks in order to provide user-friendly messages, e.g. `Objects.requireNonNull(supplier, "supplier is null");`